[
    {
        "content": "<p>in <a href=\"#narrow/stream/231634-beginners/topic/listing.20files.20in.20a.20directory/near/403250546\">this post</a> I noticed that there's an unreachable branch in a <code>when</code>:</p>\n<div class=\"codehilite\" data-code-language=\"CoffeeScript\"><pre><span></span><code><span class=\"k\">when</span><span class=\"w\"> </span><span class=\"nx\">err</span><span class=\"w\"> </span><span class=\"o\">is</span>\n<span class=\"w\">    </span><span class=\"nx\">IOError</span><span class=\"w\"> </span><span class=\"nx\">_</span><span class=\"w\"> </span><span class=\"nx\">_</span><span class=\"w\"> </span><span class=\"nf\">-&gt;</span><span class=\"w\"> </span><span class=\"s\">\"Error reading dir\"</span>\n</code></pre></div>\n<p>there won't be a tag of type <code>IOError</code> there at that point (in part because we didn't wrap the possible I/O errors in an outer tag to facilitate matching like this; whoops - we should change it from <code>IOError</code> to <code>[IOError IOError]</code>), but there's no compiler warning about it.</p>",
        "id": 403253062,
        "sender_full_name": "Richard Feldman",
        "timestamp": 1700517929
    },
    {
        "content": "<p>does anyone know if this is a bug, or if we just don't report unreachable branches like this?</p>",
        "id": 403253079,
        "sender_full_name": "Richard Feldman",
        "timestamp": 1700517941
    },
    {
        "content": "<p><a href=\"https://github.com/roc-lang/roc/issues/3692\">https://github.com/roc-lang/roc/issues/3692</a></p>",
        "id": 403274155,
        "sender_full_name": "Ayaz Hafiz",
        "timestamp": 1700528343
    },
    {
        "content": "<p><a href=\"https://github.com/roc-lang/roc/pull/3601\">https://github.com/roc-lang/roc/pull/3601</a> is a super old PR that implements the basic version to do this check</p>",
        "id": 403274188,
        "sender_full_name": "Ayaz Hafiz",
        "timestamp": 1700528357
    }
]